event_management/src/main/java/com/example/event_management/service/UserEventService.java

```diff
@@ -15,35 +19,54 @@ public class UserEventService {
     @Autowired
     UserEventRepository userEventRepository;
 
-    public void deleteByEventId(String id) {
-        userEventRepository.deleteByEventId(id);
+    @Autowired
+    EventsRepository eventsRepository;
+
+    @Test
+    @Transactional
+    public void deleteByEventId(String eventId) {
+
+        userEventRepository.deleteByEventId(eventId);
     }
 
-    public void deleteByUserId(String id) {
-        userEventRepository.deleteByUserId(id);
+    @Test
+    @Transactional
+    public void deleteByUserId(String userId) {
+
+        userEventRepository.deleteByUserId(userId);
     }
 
     public void bookAnEvent(UserEvent userEvent) {
+        Events event = eventsRepository.findById(userEvent.getEventId()).get();
+
+        if(event.getSeatsLeft()==0){
+            throw new ObjectValidationException("'seatsLeft' can not be negative");
+        }
 
         UserEvent bookEvent = userEventRepository.findByEventIdAndUserId(userEvent.getEventId(), userEvent.getUserId());
 
         if(Objects.nonNull(bookEvent)){
 
             if(bookEvent.isBooked()) {
-                bookEvent.setBooked(false);
+                event.setSeatsLeft(event.getSeatsLeft()+1);
             }else{
-                bookEvent.setBooked(true);
+                event.setSeatsLeft(event.getSeatsLeft()-1);
             }
+
+            bookEvent.setBooked(!bookEvent.isBooked());
```

Comment:
why change in value?
```suggestion
            bookEvent.setBooked(bookEvent.isBooked());
```

Reply:
bookEvent.isBooked() is taken from mysql table and so it is changing its value after booking/not booking